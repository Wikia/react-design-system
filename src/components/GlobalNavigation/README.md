User:
```js
<GlobalNavigation api={() => Promise.resolve({data: require('./mockedData').user})}/>
```

Anon:
```js
<GlobalNavigation api={() => Promise.resolve({data: require('./mockedData').anon})}/>
```

Anon with Partner Slot:
```js
<GlobalNavigation api={() => Promise.resolve({data: require('./mockedData').anonPartner})}/>
```
