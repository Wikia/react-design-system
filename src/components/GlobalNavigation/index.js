import React from 'react';
import PropTypes from 'prop-types';

import GlobalNavigationDesktop from './components/GlobalNavigationDesktop';

class GlobalNavigation extends React.Component {
  state = {
    navigationData: null,
  };

  componentDidMount() {
    this.props.api().then(
      (response) => {
        this.setState({navigationData: response.data});
      }
    );
  }

  render() {
    if (this.state.navigationData === null) {
      // loading
      return null;
    }

    return (
      <GlobalNavigationDesktop
        className={this.props.className}
        navigationData={this.state.navigationData}
      />
    );
  }
}

GlobalNavigation.propTypes = {
  /**
   * Additional class name for the component
   */
  className: PropTypes.string,
  /**
   * A function returning an AXION promise
   */
  api: PropTypes.func.isRequired,
};

GlobalNavigation.defaultProps = {
  className: '',
};

export default GlobalNavigation;
