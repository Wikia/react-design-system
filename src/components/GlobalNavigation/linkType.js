import PropTypes from 'prop-types';

const linkType = PropTypes.shape({
  type: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  // optional fields
  caption: PropTypes.string,
  edition: PropTypes.string,
  href: PropTypes.string,
  image: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.shape()),
  tracking_label: PropTypes.string,
});

export default linkType;
