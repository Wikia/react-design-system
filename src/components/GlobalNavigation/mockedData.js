export default {
  logo: {
    type: 'link-logo',
    title: 'FANDOM',
    href: 'http://fandom.wikia.com/',
    tracking_label: 'logo',
    edition: 'UK',
  },
  search: {
    url: 'http://fandom.wikia.com',
    param: 's',
    label: 'Search',
    placeholder_active: 'Placeholder Active',
    placeholder_inactive: 'Placeholder Inactive',
    tracking_label: 'search',
  },
  create_wiki: {
    type: 'link-button',
    title: 'Create a wiki',
    href: 'http://fandom.wikia.com/',
    tracking_label: 'start-a-wiki',
  },
  main_navigation: [
    {
      type: 'link-nav',
      title: 'Games',
      href: 'http://fandom.wikia.com/topic/games',
      tracking_label: 'link.games',
    },
    {
      type: 'link-nav',
      title: 'Movies',
      href: 'http://fandom.wikia.com/topic/games',
      tracking_label: 'link.movies',
    },
    {
      type: 'link-nav',
      title: 'TV',
      href: 'http://fandom.wikia.com/topic/games',
      tracking_label: 'link.tv',
    },
    {
      type: 'link-nav',
      title: 'Video',
      href: 'http://fandom.wikia.com/video',
      tracking_label: 'link.video',
    },
    {
      type: 'link-group',
      title: 'Wikis',
      items: [
        {
          type: 'link-text',
          title: 'Explore',
          href: 'http://fandom.wikia.com/explore',
          tracking_label: 'link.explore',
        },
        {
          type: 'link-text',
          title: 'Community',
          href: 'http://de.community.sandbox-content.wikia.com/wiki/Community_Deutschland',
          tracking_label: 'link.community-central',
        },
        {
          type: 'link-text',
          title: 'Fandom University',
          href: 'http://fandom.wikia.com/wiki/Fandom_University',
          tracking_label: 'link.fandom-university',
        },
        {
          type: 'link-button',
          title: 'Create a wiki',
          href: 'http://fandom.wikia.com/',
          tracking_label: 'start-a-wiki',
        },
      ],
    },
  ],
  __user: {
    avatar: 'https://static.wikia.nocookie.net/2536a38e-ab79-4d85-a5a0-16428e2582e8/scale-to-width-down/50',
    username: 'User name',
    tracking_label: 'account',
    items: [
      {
        type: 'link-text',
        title: 'View wiki profile',
        href: 'http://community.wikia.com/wiki/User:User_name',
        tracking_label: 'account.profile-wiki',
      },
      {
        type: 'link-text',
        title: 'View author profile',
        href: 'http://fandom.wikia.com/u/User_name',
        tracking_label: 'account.profile',
      },
      {
        type: 'link-logout',
        title: 'Sign-out',
        href: 'https://www.sandbox-content.wikia.com/signout?redirect=<referrer>',
        tracking_label: 'account.sign-out',
      },
    ],
  },
  anon: [
    {
      type: 'link-full-button',
      href: 'https://www.sandbox-content.wikia.com/signin?redirect=<referrer>',
      title: 'Sign-in',
      tracking_label: 'account.sign-in',
    },
    {
      type: 'link-full-button-secondary',
      href: 'https://www.sandbox-content.wikia.com/register?redirect=<referrer>',
      title: 'Register',
      tracking_label: 'account.register',
      caption: 'Don\'t have an account?',
    },
  ],
  partner_slot: {
    type: 'link-partner-slot',
    href: 'http://www.entertainweb.de/',
    image: 'https://services.wikia.com/static-assets/image/5588e692-fae8-4dc3-8db6-5f62e37fed47',
    title: 'entertainweb',
    tracking_label: 'entertainweb',
  },
};
