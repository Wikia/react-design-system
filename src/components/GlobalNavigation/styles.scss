// import variables
@import "~design-system/dist/scss/wds-functions/index.scss";
@import "~design-system/dist/scss/wds-mixins/index.scss";
@import "~design-system/dist/scss/wds-variables/index.scss";

// import wds-button
@import "~design-system/dist/scss/wds-components/_global-navigation.scss";

// hacks
.wds-global-navigation__partner-slot-image {
	max-height: 35px;
	max-width: 60px;
}
