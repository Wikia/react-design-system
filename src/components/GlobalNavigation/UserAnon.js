import React from 'react';
import PropTypes from 'prop-types';

import Link from './Link';
import linkType from './linkType';

const UserAnon = ({anonLinks}) => {
  if (!anonLinks) {
    return null;
  }

  return (
    <div className="wds-global-navigation__user-menu wds-has-shadow wds-dropdown">
      <div className="wds-dropdown__toggle">
        <div className="wds-avatar">
          <div className="wds-avatar__inner-border" />
          <svg className="wds-avatar__image">
            <use xlinkHref="#wds-avatar-icon-user" />
          </svg>
        </div>
        <svg className="wds-icon wds-icon-tiny wds-dropdown__toggle-chevron">
          <use xlinkHref="#wds-icons-dropdown-tiny" />
        </svg>
      </div>
      <div className="wds-global-navigation__dropdown-content wds-is-right-aligned wds-dropdown__content">
        <ul className="wds-list wds-has-lines-between">
          {anonLinks.map(item => (
            <li key={item.tracking_label}>
              {item.caption && (
                <div className="wds-global-navigation__user-menu-dropdown-caption">
                  {item.caption}
                </div>
              )}
              <Link
                className={`wds-is-full-width wds-button ${item.secondary && 'wds-is-secondary'}`}
                link={item}
              />
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

UserAnon.propTypes = {
  anonLinks: PropTypes.arrayOf(linkType),
};

UserAnon.defaultProps = {
  anonLinks: null,
};

export default UserAnon;
