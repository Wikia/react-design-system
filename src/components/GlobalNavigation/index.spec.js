import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import sinon from 'sinon';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import GlobalNavigation from './index';

const mockedData = {
  logo: {
    type: 'link-logo',
    title: 'FANDOM',
    href: 'http://fandom.wikia.com/',
    tracking_label: 'logo',
    edition: 'UK',
  },
  search: {
    url: 'http://fandom.wikia.com',
    param: 's',
    label: 'Search',
    placeholder_active: 'Placeholder Active',
    placeholder_inactive: 'Placeholder Inactive',
    tracking_label: 'search',
  },
  create_wiki: {
    type: 'link-button',
    title: 'Create a wiki',
    href: 'http://fandom.wikia.com/',
    tracking_label: 'start-a-wiki',
  },
  main_navigation: [
    {
      type: 'link-nav',
      title: 'Games',
      href: 'http://fandom.wikia.com/topic/games',
      tracking_label: 'link.games',
    },
    {
      type: 'link-nav',
      title: 'Movies',
      href: 'http://fandom.wikia.com/topic/games',
      tracking_label: 'link.movies',
    },
    {
      type: 'link-nav',
      title: 'TV',
      href: 'http://fandom.wikia.com/topic/games',
      tracking_label: 'link.tv',
    },
    {
      type: 'link-nav',
      title: 'Video',
      href: 'http://fandom.wikia.com/video',
      tracking_label: 'link.video',
    },
    {
      type: 'link-group',
      title: 'Wikis',
      items: [
        {
          type: 'link-text',
          title: 'Explore',
          href: 'http://fandom.wikia.com/explore',
          tracking_label: 'link.explore',
        },
        {
          type: 'link-text',
          title: 'Community',
          href: 'http://de.community.sandbox-content.wikia.com/wiki/Community_Deutschland',
          tracking_label: 'link.community-central',
        },
        {
          type: 'link-text',
          title: 'Fandom University',
          href: 'http://fandom.wikia.com/wiki/Fandom_University',
          tracking_label: 'link.fandom-university',
        },
        {
          type: 'link-button',
          title: 'Create a wiki',
          href: 'http://fandom.wikia.com/',
          tracking_label: 'start-a-wiki',
        },
      ],
    },
  ],
};

test('GlobalNavigation should work with axios', () => {
  const mock = new MockAdapter(axios);
  mock.onGet('/data').reply(200, mockedData);

  const api = () => axios.get('/data');

  const wrapper = mount(
    <GlobalNavigation api={api} />
  );
  expect(wrapper).toMatchSnapshot();
});
