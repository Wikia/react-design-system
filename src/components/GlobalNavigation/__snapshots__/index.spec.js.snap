// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalNavigation should work with axios 1`] = `
<GlobalNavigation
  api={[Function]}
  className=""
/>
`;
